fastify.js

```diff
@@ -917,6 +914,26 @@ function fastify (options) {
     this[kGenReqId] = reqIdGenFactory(this[kOptions].requestIdHeader, func)
     return this
   }
+
+  function addHttpMethod (method, { hasBody = false } = {}) {
+    if (typeof method !== 'string' || http.METHODS.indexOf(method) === -1) {
+      throw new FST_ERR_ROUTE_METHOD_INVALID()
+    }
+
+    if (hasBody === true) {
+      this[kSupportedHTTPMethods].bodywith.add(method)
+      this[kSupportedHTTPMethods].bodyless.delete(method)
+    } else {
+      this[kSupportedHTTPMethods].bodywith.delete(method)
+      this[kSupportedHTTPMethods].bodyless.add(method)
+    }
+
+    this[method.toLowerCase()] = function (url, options, handler) {
+      return router.prepareRoute.call(this, { method, url, options, handler })
+    }
```

Comment:
Will it be better to use `decorate`? 
It can prevent overriding the same function again?

```suggestion
    const _method = method.toLowerCase()
    if (!this.hasDecorator(_method)) {
      this.decorate(_method, function(url, options, handler) {
        router.prepareRoute.call(this, { method, url, options, handler })
      })
    }
    this[method.toLowerCase()] = function (url, options, handler) {
      return router.prepareRoute.call(this, { method, url, options, handler })
    }
```